src/app/InteractionModelEngine.cpp

```diff
@@ -205,34 +205,48 @@ CHIP_ERROR InteractionModelEngine::OnInvokeCommandRequest(Messaging::ExchangeCon
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR InteractionModelEngine::OnReadInitialRequest(Messaging::ExchangeContext * apExchangeContext,
-                                                        const PayloadHeader & aPayloadHeader,
-                                                        System::PacketBufferHandle && aPayload,
-                                                        ReadHandler::InteractionType aInteractionType,
-                                                        Protocols::InteractionModel::Status & aStatus)
+CHIP_ERROR InteractionModelEngine::CheckWhetherKeepExistingSubscription(Messaging::ExchangeContext * apExchangeContext,
+                                                                        System::PacketBufferHandle && aPayload)
 {
-    ChipLogDetail(InteractionModel, "Received %s request",
-                  aInteractionType == ReadHandler::InteractionType::Subscribe ? "Subscribe" : "Read");
+    bool keepSubscriptions = true;
+    System::PacketBufferTLVReader reader;
+    reader.Init(aPayload.Retain());
```

Comment:
Why retain here? Caller handed us an rvalue... just:
```suggestion
    reader.Init(std::move(aPayload));
```

Reply:
Done, thanks